lib/extensions/ar_to_model_hash.rb

```diff
@@ -110,7 +110,14 @@ def to_model_hash_build_preload(options)
     columns  = ((options && options[:columns]) || [])
     includes = ((options && options[:include]) || {})
 
-    result  = columns.select { |c| self.class.virtual_column?(c) }
+    result = columns.select do |c|
+      next unless self.class.virtual_column?(c)
+
+      # preload the virtual field's uses if they're not already included
+      uses = Array.wrap(self.class.virtual_field(c).uses)
+      (uses - includes.keys).empty?
```

Comment:
The real problem the includes has a column on a different table that match a virtual column in the source table (vms).  We include the hardware table having a column that matches a same named virtual column on VmOrTemplate.
I need to think about it so more but we should fix this differently.